### PR TITLE
Improvement/plugin

### PR DESCRIPTION
**Pull request by soupette**
- Update `request.js` helper to handle serverRestart watcher (prepare when route `_health`is ready.
- Remove all server restart timer in plugins' `sagas`
- Created a global `BlockerComponent` that is rendered instead of a plugin (if the plugin's requirements aren't met)
- Update plugin rendering and allow to pass a `customBlockerComponent`
- Update stateless component generator with ES6 function
- Add a button loader to `settings-manager` plugin which informs the user that the server will restart
- Add number types translations to content-type view in `content-type-builder` plugin

### CODE REVIEW

File: packages/strapi-plugin-content-type-builder/admin/src/containers/Form/sagas.js [hunk 1]
```diff
@@ -29,21 +29,17 @@ export function* editContentType() {
       method: 'PUT',
       body,
     };
+    const shouldWatchServerRestart = true;
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-content-type-builder/admin/src/containers/Form/sagas.js]: Why are you declaring this variable if the value is always set to true?

File: packages/strapi-plugin-content-type-builder/admin/src/containers/ModelPage/sagas.js [hunk 2]
```diff
@@ -58,34 +58,28 @@ export function* submitChanges() {
     const method = modelName === body.name ? 'POST' : 'PUT';
     const baseUrl = '/content-type-builder/models/';
     const requestUrl = method === 'POST' ? baseUrl : `${baseUrl}${body.name}`;
-
     const opts = { method, body };
+    const shouldWatchServerRestart = true;
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-content-type-builder/admin/src/containers/ModelPage/sagas.js]: Same comment as above...

File: packages/strapi-plugin-settings-manager/admin/src/components/Button/index.js [hunk 3]
```diff
@@ -16,9 +17,23 @@ class Button extends React.Component {
...
+
+    if (this.props.loader) {
+      return (
+        <button
+          type="button"
+          className={cn(styles.loader, styles.primary)}
+          disabled
+        >
+          <div className={styles.saving}>
+            <p></p><p></p><p></p>
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-settings-manager/admin/src/components/Button/index.js]: What's that?

File: packages/strapi-plugin-content-type-builder/admin/src/containers/Form/sagas.js (hunk 1 above)
Comment (soupette) [reply packages/strapi-plugin-content-type-builder/admin/src/containers/Form/sagas.js]: It's for information purpose

File: packages/strapi-plugin-content-type-builder/admin/src/containers/ModelPage/sagas.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-plugin-content-type-builder/admin/src/containers/ModelPage/sagas.js]: It's for information purpose

File: packages/strapi-plugin-settings-manager/admin/src/components/Button/index.js (hunk 3 above)
Comment (soupette) [reply packages/strapi-plugin-settings-manager/admin/src/components/Button/index.js]: To insert fontawesome dot icon

**Pull request merged**